com.unity.sg2/Editor/GraphUI/GraphElements/Views/ShaderGraphViewSelection.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Linq;
+using UnityEditor.GraphToolsFoundation.Overdrive;
+using UnityEditor.GraphToolsFoundation.Overdrive.BasicModel;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    public class ShaderGraphViewSelection : GraphViewSelection
+    {
+        public ShaderGraphViewSelection(GraphView view, GraphModelStateComponent graphModelState, SelectionStateComponent selectionState)
+            : base(view, graphModelState, selectionState)
+        {
+
+        }
+
+        /// <summary>
+        /// Adds items related to the selection to the contextual menu.
+        /// </summary>
+        /// <param name="evt">The contextual menu event.</param>
+        public override void BuildContextualMenu(ContextualMenuPopulateEvent evt)
+        {
+            if (evt.menu.MenuItems().Count > 0)
+                evt.menu.AppendSeparator();
+
+            if (GetSelection().Any(model => model is INodeModel))
```

Comment:
This is to get rid of some of the extra GTF functionality that gets added on by default as we don't currently really support it and its not a part of the parity target with old SG.